### ISSUE TITLE
Warnings when building with clang on Mac OS X

### ISSUE THREAD

**Issue by njh** [libcheck/check#58]
Hi,

I have just started using libcheck again for a new project. While the tests are working file, I am getting complier warnings with clang:

    /usr/local/Cellar/check/0.10.0/include/check.h:454:75: error: token pasting of ',' and __VA_ARGS__ is a GNU extension
      [-Werror,-Wgnu-zero-variadic-macro-arguments]
     _ck_assert_failed(__FILE__, __LINE__, "Assertion '"#expr"' failed" , ## __VA_ARGS__, NULL)

The hacky way to work round this is using the ```-Wno-gnu-zero-variadic-macro-arguments``` compiler flag. But I think it would be better to not have to do this.

check: stable 0.10.0 (bottled)
Clang: Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin14.5.0

Also slightly surprised nobody else seems to have reported this.

nick.

**Comment by brarcher** [comment]
I think that the following pull request attempts to resolve this:

https://github.com/libcheck/check/pull/40

However, the proposed changes do not yet test successfully when using MSVC on Windows. If you happen to have access to Windows and want to take a stab at it, I would be grateful. (:  Though, if you are reporting this from OSX, you may not have Windows with MSVC installed.

**Comment by njh** [comment]
Yeah, sorry I don't have access to Windows.

**Comment by iasonov** [comment]
Same issue on my Mac (M1). I have Parallels (so I can try test proposed changes with MSVC on Win 11).

check: 0.15.2
gcc:
Apple clang version 15.0.0 (clang-1500.3.9.4)
target: arm64-apple-darwin23.4.0
